Error when  'make LibCarla'

Hi,

I'm using Ubuntu 18.04.1 LTS.
And I got this list of errors shown below: (Full log attached.) 
[log.txt](https://github.com/carla-simulator/carla/files/2481649/log.txt)


BuildLibCarla.sh: Building LibCarla "Server" configuration.
[1/51] Building CXX object LibCarla/cm.../__/__/source/carla/client/Image.cpp.o
FAILED: LibCarla/cmake/server/CMakeFiles/carla_server.dir/__/__/source/carla/client/Image.cpp.o 
/usr/bin/clang++-5.0  -DBOOST_ERROR_CODE_HEADER_ONLY -I../../LibCarla/cmake/../source -I../boost-1.67.0-install/include -I../rpclib-2.2.1-libcxx-install/include -std=c++14 -pthread -fPIC -Werror -Wall -Wextra -Wno-missing-braces -stdlib=libc++ -I/home/ic/UnrealEngine_4.19/carla/Build/llvm-5.0-install/include/c++/v1 -std=c++14 -pthread -fPIC -Werror -Wall -Wextra -Wno-missing-braces -stdlib=libc++ -I/home/ic/UnrealEngine_4.19/carla/Build/llvm-5.0-install/include/c++/v1   -O3 -DNDEBUG -MD -MT LibCarla/cmake/server/CMakeFiles/carla_server.dir/__/__/source/carla/client/Image.cpp.o -MF LibCarla/cmake/server/CMakeFiles/carla_server.dir/__/__/source/carla/client/Image.cpp.o.d -o LibCarla/cmake/server/CMakeFiles/carla_server.dir/__/__/source/carla/client/Image.cpp.o -c ../../LibCarla/source/carla/client/Image.cpp
In file included from ../../LibCarla/source/carla/client/Image.cpp:8:
In file included from ../../LibCarla/cmake/../source/carla/client/Image.h:9:
../../LibCarla/cmake/../source/carla/Buffer.h:204:42: error: no member named 'is_const_buffer_sequence' in namespace 'boost::asio'
    typename std::enable_if<boost::asio::is_const_buffer_sequence<T>::value>::type
                            ~~~~~~~~~~~~~^
../../LibCarla/cmake/../source/carla/Buffer.h:204:67: error: 'T' does not refer to a value
    typename std::enable_if<boost::asio::is_const_buffer_sequence<T>::value>::type
                                                                  ^
../../LibCarla/cmake/../source/carla/Buffer.h:203:24: note: declared here
    template <typename T>
                       ^
../../LibCarla/cmake/../source/carla/Buffer.h:204:71: error: member 'value' declared as a template
    typename std::enable_if<boost::asio::is_const_buffer_sequence<T>::value>::type
                                                                      ^
../../LibCarla/cmake/../source/carla/Buffer.h:204:76: error: expected ';' at end of declaration list
    typename std::enable_if<boost::asio::is_const_buffer_sequence<T>::value>::type
                                                                           ^
                                                                           ;
../../LibCarla/cmake/../source/carla/Buffer.h:97:9: error: use of undeclared identifier 'ReuseThisBuffer'
        ReuseThisBuffer();
        ^
5 errors generated.

Thank you for your help.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 ## Latest changes
 
+  * Upgraded Boost to 1.69.0
   * Added point transformation functionality for LibCarla and PythonAPI
   * Added "sensor_tick" attribute to sensors (cameras and lidars) to specify the capture rate in seconds
   * Added "get_forward_vector()" to rotation and transform, retrieves the unit vector on the rotation's X-axis

--- a/Docs/build_system.md
+++ b/Docs/build_system.md
@@ -28,7 +28,7 @@ Get and compile dependencies
 
   * llvm-5.0 (libc++ and libc++abi)
   * rpclib-2.2.1 (twice, with libstdc++ and libc++)
-  * boost-1.67 (headers only)
+  * boost-1.69 (headers only)
   * googletest-1.8.0 (with libc++)
 
 #### LibCarla

--- a/LibCarla/source/carla/image/BoostGil.h
+++ b/LibCarla/source/carla/image/BoostGil.h
@@ -13,7 +13,7 @@
 #  pragma clang diagnostic ignored "-Wunused-local-typedef"
 #endif
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil.hpp>
 
 #if defined(__clang__)
 #  pragma clang diagnostic pop

--- a/LibCarla/source/carla/image/ImageIOConfig.h
+++ b/LibCarla/source/carla/image/ImageIOConfig.h
@@ -47,15 +47,23 @@
 #  ifndef int_p_NULL
 #    define int_p_NULL (int*)NULL
 #  endif // int_p_NULL
-#  include <boost/gil/extension/io/png_io.hpp>
+#  if defined(__clang__)
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wignored-qualifiers"
+#    pragma clang diagnostic ignored "-Wparentheses"
+#  endif
+#    include <boost/gil/extension/io/png.hpp>
+#  if defined(__clang__)
+#    pragma clang diagnostic pop
+#  endif
 #endif
 
 #if LIBCARLA_IMAGE_WITH_JPEG_SUPPORT == true
-#  include <boost/gil/extension/io/jpeg_io.hpp>
+#  include <boost/gil/extension/io/jpeg.hpp>
 #endif
 
 #if LIBCARLA_IMAGE_WITH_TIFF_SUPPORT == true
-#  include <boost/gil/extension/io/tiff_io.hpp>
+#  include <boost/gil/extension/io/tiff.hpp>
 #endif
 
 #if defined(__clang__)
@@ -85,6 +93,11 @@ namespace io {
 
 namespace detail {
 
+  template <typename ViewT, typename IOTag>
+  struct is_write_supported {
+    static constexpr bool value = boost::gil::is_write_supported<typename boost::gil::get_pixel_type<ViewT>::type, IOTag>::value;
+  };
+
   struct io_png {
 
     static constexpr bool is_supported = has_png_support();
@@ -102,12 +115,12 @@ namespace detail {
 
     template <typename Str, typename ImageT>
     static void read_image(Str &&in_filename, ImageT &image) {
-      boost::gil::png_read_and_convert_image(std::forward<Str>(in_filename), image);
+      boost::gil::read_and_convert_image(std::forward<Str>(in_filename), image, boost::gil::png_tag());
     }
 
     template <typename Str, typename ViewT>
     static void write_view(Str &&out_filename, const ViewT &view) {
-      boost::gil::png_write_view(std::forward<Str>(out_filename), view);
+      boost::gil::write_view(std::forward<Str>(out_filename), view, boost::gil::png_tag());
     }
 
 #endif // LIBCARLA_IMAGE_WITH_PNG_SUPPORT
@@ -131,21 +144,22 @@ namespace detail {
 
     template <typename Str, typename ImageT>
     static void read_image(Str &&in_filename, ImageT &image) {
-      boost::gil::jpeg_read_image(std::forward<Str>(in_filename), image);
+      boost::gil::read_image(std::forward<Str>(in_filename), image, boost::gil::jpeg_tag());
     }
 
     template <typename Str, typename ViewT>
-    static typename std::enable_if<boost::gil::jpeg_write_support<ViewT>::is_supported>::type
+    static typename std::enable_if<is_write_supported<ViewT, boost::gil::jpeg_tag>::value>::type
     write_view(Str &&out_filename, const ViewT &view) {
-      using namespace boost::gil;
-      jpeg_write_view(std::forward<Str>(out_filename), view);
+      boost::gil::write_view(std::forward<Str>(out_filename), view, boost::gil::jpeg_tag());
     }
 
     template <typename Str, typename ViewT>
-    static typename std::enable_if<!boost::gil::jpeg_write_support<ViewT>::is_supported>::type
+    static typename std::enable_if<!is_write_supported<ViewT, boost::gil::jpeg_tag>::value>::type
     write_view(Str &&out_filename, const ViewT &view) {
-      using namespace boost::gil;
-      jpeg_write_view(std::forward<Str>(out_filename), color_converted_view<rgb8_pixel_t>(view));
+      boost::gil::write_view(
+          std::forward<Str>(out_filename),
+          boost::gil::color_converted_view<boost::gil::rgb8_pixel_t>(view),
+          boost::gil::jpeg_tag());
     }
 
 #endif // LIBCARLA_IMAGE_WITH_JPEG_SUPPORT
@@ -168,21 +182,22 @@ namespace detail {
 
     template <typename Str, typename ImageT>
     static void read_image(Str &&in_filename, ImageT &image) {
-      boost::gil::tiff_read_and_convert_image(std::forward<Str>(in_filename), image);
+      boost::gil::read_and_convert_image(std::forward<Str>(in_filename), image, boost::gil::tiff_tag());
     }
 
     template <typename Str, typename ViewT>
-    static typename std::enable_if<boost::gil::tiff_write_support<ViewT>::is_supported>::type
+    static typename std::enable_if<is_write_supported<ViewT, boost::gil::tiff_tag>::value>::type
     write_view(Str &&out_filename, const ViewT &view) {
-      using namespace boost::gil;
-      tiff_write_view(std::forward<Str>(out_filename), view);
+      boost::gil::write_view(std::forward<Str>(out_filename), view, boost::gil::tiff_tag());
     }
 
     template <typename Str, typename ViewT>
-    static typename std::enable_if<!boost::gil::tiff_write_support<ViewT>::is_supported>::type
+    static typename std::enable_if<!is_write_supported<ViewT, boost::gil::tiff_tag>::value>::type
     write_view(Str &&out_filename, const ViewT &view) {
-      using namespace boost::gil;
-      tiff_write_view(std::forward<Str>(out_filename), color_converted_view<rgb8_pixel_t>(view));
+      boost::gil::write_view(
+          std::forward<Str>(out_filename),
+          boost::gil::color_converted_view<boost::gil::rgb8_pixel_t>(view),
+          boost::gil::tiff_tag());
     }
 
 #endif // LIBCARLA_IMAGE_WITH_TIFF_SUPPORT

--- a/LibCarla/source/compiler/disable-ue4-macros.h
+++ b/LibCarla/source/compiler/disable-ue4-macros.h
@@ -15,16 +15,6 @@
 #  define BOOST_NO_EXCEPTIONS
 #endif // BOOST_NO_EXCEPTIONS
 
-// Suppress clang warning.
-#if defined(__clang__)
-#  ifndef __cpp_coroutines
-#    define __cpp_coroutines 0
-#  endif // __cpp_coroutines
-#  ifndef __cpp_noexcept_function_type
-#    define __cpp_noexcept_function_type 0
-#  endif // __cpp_noexcept_function_type
-#endif // defined(__clang__)
-
 namespace boost {
 
   static inline void throw_exception(const std::exception &e) {
@@ -56,6 +46,8 @@ namespace boost {
 #if defined(__clang__)
 #  pragma clang diagnostic push
 #  pragma clang diagnostic ignored "-Wmissing-braces"
+#  pragma clang diagnostic ignored "-Wunusable-partial-specialization"
+#  pragma clang diagnostic ignored "-Wundef"
 #endif
 
 #pragma push_macro("TEXT")

--- a/Util/BuildTools/Setup.sh
+++ b/Util/BuildTools/Setup.sh
@@ -71,7 +71,7 @@ unset LLVM_BASENAME
 # -- Get boost includes --------------------------------------------------------
 # ==============================================================================
 
-BOOST_VERSION=1.68.0
+BOOST_VERSION=1.69.0
 BOOST_BASENAME="boost-${BOOST_VERSION}"
 
 BOOST_INCLUDE=${PWD}/${BOOST_BASENAME}-install/include

--- a/Util/BuildTools/Setup.sh
+++ b/Util/BuildTools/Setup.sh
@@ -71,7 +71,8 @@ unset LLVM_BASENAME
 # -- Get boost includes --------------------------------------------------------
 # ==============================================================================
 
-BOOST_BASENAME=boost-1.67.0
+BOOST_VERSION=1.67.0
+BOOST_BASENAME="boost-${BOOST_VERSION}"
 
 BOOST_INCLUDE=${PWD}/${BOOST_BASENAME}-install/include
 BOOST_LIBPATH=${PWD}/${BOOST_BASENAME}-install/lib
@@ -83,12 +84,11 @@ else
   rm -Rf ${BOOST_BASENAME}-source
 
   log "Retrieving boost."
-  wget https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.gz
+  wget "https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source/boost_${BOOST_VERSION//./_}.tar.gz"
   log "Extracting boost."
-  tar -xzf boost_1_67_0.tar.gz
+  tar -xzf ${BOOST_BASENAME//[-.]/_}.tar.gz
   mkdir -p ${BOOST_BASENAME}-install/include
-  mv boost_1_67_0 ${BOOST_BASENAME}-source
-  # rm -Rf boost_1_67_0
+  mv ${BOOST_BASENAME//[-.]/_} ${BOOST_BASENAME}-source
 
   pushd ${BOOST_BASENAME}-source >/dev/null
 
@@ -114,13 +114,13 @@ else
   ./b2 toolset="${BOOST_TOOLSET}" cxxflags="${BOOST_CFLAGS}" --prefix="../${BOOST_BASENAME}-install" -j ${CARLA_BUILD_CONCURRENCY} stage release
   ./b2 toolset="${BOOST_TOOLSET}" cxxflags="${BOOST_CFLAGS}" --prefix="../${BOOST_BASENAME}-install" -j ${CARLA_BUILD_CONCURRENCY} install
   ./b2 toolset="${BOOST_TOOLSET}" cxxflags="${BOOST_CFLAGS}" --prefix="../${BOOST_BASENAME}-install" -j ${CARLA_BUILD_CONCURRENCY} --clean-all
-  
+
   # Get rid of  python2 build artifacts completely & do a clean build for python3
   popd >/dev/null
   rm -Rf ${BOOST_BASENAME}-source
-  tar -xzf boost_1_67_0.tar.gz
+  tar -xzf ${BOOST_BASENAME//[-.]/_}.tar.gz
   mkdir -p ${BOOST_BASENAME}-install/include
-  mv boost_1_67_0 ${BOOST_BASENAME}-source
+  mv ${BOOST_BASENAME//[-.]/_} ${BOOST_BASENAME}-source
   pushd ${BOOST_BASENAME}-source >/dev/null
 
   py3="/usr/bin/env python3"
@@ -145,6 +145,7 @@ else
   popd >/dev/null
 
   rm -Rf ${BOOST_BASENAME}-source
+  rm ${BOOST_BASENAME//[-.]/_}.tar.gz
 
 fi
 
@@ -178,7 +179,7 @@ else
 
   log "Building rpclib with libc++."
 
-  # rpclib does not use any cmake 3.9 feature. 
+  # rpclib does not use any cmake 3.9 feature.
   # As cmake 3.9 is not standard in Ubuntu 16.04, change cmake version to 3.5
   sed -i s/"3.9.0"/"3.5.0"/g ${RPCLIB_BASENAME}-source/CMakeLists.txt
 

--- a/Util/BuildTools/Setup.sh
+++ b/Util/BuildTools/Setup.sh
@@ -71,7 +71,7 @@ unset LLVM_BASENAME
 # -- Get boost includes --------------------------------------------------------
 # ==============================================================================
 
-BOOST_VERSION=1.67.0
+BOOST_VERSION=1.68.0
 BOOST_BASENAME="boost-${BOOST_VERSION}"
 
 BOOST_INCLUDE=${PWD}/${BOOST_BASENAME}-install/include

--- a/Util/InstallersWin/install_boost.bat
+++ b/Util/InstallersWin/install_boost.bat
@@ -42,7 +42,7 @@ if [%B_TOOLSET%] == [] set B_TOOLSET=msvc-14.1
 rem If is not set set the number of parallel jobs to the number of CPU threads
 if [%NUMBER_OF_ASYNC_JOBS%] == [] set NUMBER_OF_ASYNC_JOBS=%NUMBER_OF_PROCESSORS%
 
-set B_VERSION=boost-1.67.0
+set B_VERSION=boost-1.69.0
 set B_SRC=boost-src
 set B_SRC_DIR=%BUILD_DIR%%B_SRC%
 set B_INSTALL=boost-install

